### ISSUE TITLE
Explore Bessel.jl for Chebychev coeffs

### ISSUE THREAD

**Issue by goerz** [JuliaQuantumControl/QuantumPropagators.jl#74]
This might make the calculation of coefficients significantly faster:

See https://discourse.julialang.org/t/matrix-exponential-vector-for-various-scales/112022/17

**Comment by Fe-r-oz** [comment]
Do we need to normalize the spellings of `Chebychev` to [Chebyshev](https://www.sciencedirect.com/topics/mathematics/chebyshev-polynomial) in `QuantumPropagators.jl `?

**Comment by goerz** [comment]
No, I always spell it `Chebychev`, and that will be the canonical spelling inside this package.

**Comment by Fe-r-oz** [comment]
Alrighty!

I am excited to explore `Bessel.jl `for `chebychev coeff`! 

As you mentioned we have to use` Bessel.jl` instead of (`besselj)` from `SpecialFunctions.jl`, as it is  typically 2-12x faster than `specialfunction.jl` implementations.  Bessel.jl also have same names etc. `besselj `

![Screenshot from 2024-05-13 08-22-16](https://github.com/JuliaQuantumControl/QuantumPropagators.jl/assets/93876775/68c1db2b-8f2d-474c-bdc5-69c4bb5750fd)

> if I recall correctly, the regular exponential just uses the modified Bessel function (besseli instead of bessel) and you kill the i [in the definition of c](https://github.com/JuliaQuantumControl/QuantumPropagators.jl/blob/6a1669ecc9a5ffb32976cda706ccb2db1eae89bc/src/cheby.jl#L233) and [in the final phase factor](https://github.com/JuliaQuantumControl/QuantumPropagators.jl/blob/6a1669ecc9a5ffb32976cda706ccb2db1eae89bc/src/cheby.jl#L276)

It seems that regular exponential uses `besselj `instead of `besseli,`, so there might be no need to kill i in c and phase factor, right?  

![Screenshot from 2024-05-13 08-31-12](https://github.com/JuliaQuantumControl/QuantumPropagators.jl/assets/93876775/b31952fc-b238-4f97-afef-d4636aea6203)

**Comment by goerz** [comment]
To really "explore" this would require proper benchmarking in realistic settings. I have a WIP benchmarking repository that isn't quite ready to be pushed. Since the coefficients can typically be reused a lot, they are very unlikely to be a bottleneck in any optimization. Still, at some point, it would be nice to quantify it a bit more. This issue is more of a note to myself to look at this when my benchmarking code is in better shape. I don't think looking into this would be a good use of your time.

**Comment by Fe-r-oz** [comment]
Sounds good! I enjoyed your talk at Yao Community Call, that was comprehensive! 

Looking forward to the workshop tutorial!
